server/sonar-web/src/main/js/apps/issues/components/AppContainer.js

```diff
@@ -36,13 +39,28 @@ const mapStateToProps = (state, ownProps) => ({
   currentUser: getCurrentUser(state)
 });
 
+const fetchIssueOrganizations = issues => dispatch => {
+  if (!issues.length) {
+    return Promise.resolve();
+  }
+
+  const organizationKeys = uniq(issues.map(issue => issue.organization));
+  return getOrganizations(organizationKeys).then(
+    response => dispatch(receiveOrganizations(response.organizations)),
+    onFail(dispatch)
+  );
+};
+
 const fetchIssues = (query: Query) => (dispatch: Dispatch<*>) =>
-  searchIssues({ ...query, additionalFields: '_all' }).then(response => {
-    const parsedIssues = response.issues.map(issue =>
-      parseIssueFromResponse(issue, response.components, response.users, response.rules)
-    );
-    return { ...response, issues: parsedIssues };
-  }, onFail(dispatch));
+  searchIssues({ ...query, additionalFields: '_all' })
+    .then(response => {
+      const parsedIssues = response.issues.map(issue =>
+        parseIssueFromResponse(issue, response.components, response.users, response.rules)
+      );
+      return { ...response, issues: parsedIssues };
+    })
+    .then(response => fetchIssueOrganizations(response.issues)(dispatch).then(() => response))
```

Comment:
Ok I can do that, but then I need to remove the typings for `dispatch` or update the `Dispatch` definition to make it work. Because flow complains that `.then` doesn't exists on `dispatch`.